### PR TITLE
[Refactor] CollectionUtils, Iterators, and String methods to core lib

### PR DESCRIPTION
**Pull request by nknize**
This commit refactors the `CollectionUtils`, and `Iterators` utility classes to the core and common libraries, respectively. Several other base `Strings` utility methods are also refactored to the core `Strings` utility class. This is done to incrementally move the remaining Strings methods in server to the core String utility.

relates #5910
relates #8110

dependency of #9103

### CODE REVIEW

**Comment by nknize** [comment]
We really need to get these Remote Store tests stabilized.

**Comment by nknize** [comment]
Another remote store failure 😠 

Refiring

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.ReplicaToPrimaryPromotionIT.testFailoverWhileIndexing" -Dtests.seed=A1FA205B5E846428 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-CA -Dtests.timezone=Etc/GMT+2 -Druntime.java=20
```

**Review by saratvemulapalli — APPROVED** [review]

File: libs/core/src/main/java/org/opensearch/core/common/Strings.java [hunk 1]
```diff
@@ -35,18 +36,6 @@ public class Strings {
...
     public static final String[] EMPTY_ARRAY = new String[0];
 
-    /**
-     * Split the specified string by commas to an array.
-     *
-     * @param s the string to split
-     * @return the array of split values
-     * @see String#split(String)
-     */
-    public static String[] splitStringByCommaToArray(final String s) {
```
Comment (andrross) [inline libs/core/src/main/java/org/opensearch/core/common/Strings.java:45]: Did the `splitStringByCommaToArray` and `delimitedListToStringArray` methods just move to a different location within the same class file?

**Review by andrross — APPROVED** [review]

File: libs/core/src/main/java/org/opensearch/core/common/Strings.java (hunk 1 above)
Comment (nknize) [reply libs/core/src/main/java/org/opensearch/core/common/Strings.java:45]: Yes, that's correct

**Comment by nknize** [comment]
Another remote store failure:

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.RemoteRestoreSnapshotIT.testRestoreShallowCopySnapshotWithDifferentRepo" -Dtests.seed=C2462D8D2A5CB1F2 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=lt -Dtests.timezone=Asia/Phnom_Penh -Druntime.java=20
```

**Pull request merged**